app/components/memorable_date_component.rb

```diff
@@ -0,0 +1,154 @@
+#
+# The errors for this component are configurable via +error_messages+, and you may
+# include custom min/max validations via +range_errors+.
+class MemorableDateComponent < BaseComponent
+  attr_reader :name, :month, :day, :year, :required, :hint, :label, :form
+
+  alias_method :f, :form
+
+  ##
+  # @param [String] name Field name for hash containing month, day, and year
+  # @param [String] hint Additional Hint to show to the user
+  # @param [String] label Label for field
+  # @param [String] form Form that this field belongs to
+  # @param [String] month Starting value for month
+  # @param [String] day Starting value for day
+  # @param [String] year Starting value for year
+  # @param [String] required Whether this field is required
+  # @param [Date,DateTime] min Minimum allowed date, inclusive
+  # @param [Date,DateTime] max Maximum allowed date, inclusive
+  # @param [Hash<Symbol,String>] error_messages Array of mappings of error states to messages
+  # @param [Array<Hash] range_errors Array of custom range errors
+  # @option range_errors [Date,DateTime] :min Minimum value for range check
+  # @option range_errors [Date,DateTime] :max Maximum value for range check
+  # @option range_errors [String] :message Error message to display if range check fails
+  def initialize(
+    name:, hint:, label:, form:,
+    month: nil,
+    day: nil,
+    year: nil,
+    required: false,
+    min: nil,
+    max: nil,
+    error_messages: {},
+    range_errors: [],
+    **_tag_options
+  )
+    @name = name
+    @month = month
+    @day = day
+    @year = year
+    @required = required
+    @min = min
+    @max = max
+    @hint = hint
+    @label = label
+    @form = form
+    @tag_options = []
+    @error_messages = error_messages
+    @range_errors = range_errors
+  end
+
+  # Get error messages to be provided to the component.
+  # Includes both a hash lookup for general error messages
+  # and an array lookup for custom range error messages.
+  def error_messages
+    {
+      'error_messages' => generate_error_messages(label, @min, @max, @error_messages),
+      'range_errors' => @range_errors.map do |err|
+        new_err = {
+          message: err[:message],
+        }
+        new_err[:min] = convert_date err[:min] if !err[:min].blank?
+        new_err[:max] = convert_date err[:max] if !err[:max].blank?
+        new_err
+      end,
+    }
+  end
+
+  # Get min date as a string like 1892-01-23
+  def min
+    convert_date @min
+  end
+
+  # Get max date as a string like 1892-01-23
+  def max
+    convert_date @max
+  end
+
+  private
+
+  # Convert a Date or DateTime to a string like 1892-01-23
+  def convert_date date
+    date.strftime('%Y-%m-%d') if date.instance_of?(Date) || date.instance_of?(DateTime)
+  end
+
+  # Convert a date or DateTime to a long-form localized date string
+  def i18n_long_format date
+    if date.instance_of?(Date) || date.instance_of?(DateTime)
+      I18n.l(date, format: t('date.formats.long'))
```

Comment:
the `format` argument should be the symbol name I think?
```suggestion
      I18n.l(date, format: :long)
```
example: https://github.com/18F/identity-idp/blob/12702b5110582c3deca72e454d8c94d4e5cadc0b/app/mailers/user_mailer.rb#L219

Reply:
I originally did that, but I got this test failure:

```
Failures:

  1) I18n does not have unused keys
     Failure/Error:
       expect(unused_keys).to(
         be_empty,
         "#{unused_keys.leaves.count} unused i18n keys, run `i18n-tasks unused' to show them",
       )
     
       3 unused i18n keys, run `i18n-tasks unused' to show them
     # ./spec/i18n_spec.rb:79:in `block (2 levels) in <top (required)>'
```

Reply:
huh, maybe we could add in an `# i18n-tasks use` style comment